### PR TITLE
enable tide merging for "istio-private"

### LINKED ISSUE #2135: Enable Tide for "istio-private"
Enable Tide for the `istio-private` GitHub organization.

### PR DESCRIPTION
**Pull request by clarketm**
fixes #2135

/hold for verification of `hidden_repos`.

cc @howardjohn

### CODE REVIEW

**Comment by clarketm** [comment]
After looking at the code, speaking with @cjwagner, learning of accounts of this in production, and performing some *functional* tests is an isolated environment, this is a *safe* change.

- [x] I suggest we sync Istio GitHub labels to `istio-private` before merging, otherwise it will be difficult/impossible for a user to block a merge with one of the specified labels.

**Comment by cjwagner** [comment]
> /hold for verification of `hidden_repos`.

@clarketm FYI, the slash commands provided by Prow plugins must be specified at the start of their own line. Only parameters should be on the same line as the command for commands that accept parameters.

**Comment by clarketm** [comment]
> > /hold for verification of `hidden_repos`.
> 
> @clarketm FYI, the slash commands provided by Prow plugins must be specified at the start of their own line. Only parameters should be on the same line as the command for commands that accept parameters.


Got it, thanks! This was more for semantics in this case, as I added the label manually. But it is good info to know!

**Review by howardjohn — APPROVED** [review]

File: prow/config.yaml
Comment (cjwagner) [inline prow/config.yaml:318]: This config will merge any PR in the org as soon as the tests are passing if there isn't a `do-not-merge/*` label.  In other words Tide won't wait for code review.

If you include `reviewApprovedRequired: true`, Tide will wait for an approve review before merging. However, Tide uses the search token `review: approved` to search for approved PRs. I don't think that it matches PRs that have only been approved by non-CODEOWNERS though so until we deal with the non-applicable CODEOWNERS file we probably can't use that.

If we don't want to worry about the CODEOWNERS problem it might be easiest to use the `lgtm` plugin configured to respect github approve reviews and have Tide require that label?

**Review by cjwagner — CHANGES_REQUESTED** [review]

File: prow/config.yaml
Comment (clarketm) [reply prow/config.yaml:318]: > This config will merge any PR in the org as soon as the tests are passing if there isn't a `do-not-merge/*` label. In other words Tide won't wait for code review. If you include `reviewApprovedRequired: true`, ...

Good catch, I updated to include `reviewApprovedRequired: true`.

> Tide will wait for an approve review before merging. However, Tide uses the search token review: approved to search for approved PRs. I don't think that it matches PRs that have only been approved by non-CODEOWNERS though so until we deal with the non-applicable CODEOWNERS file we probably can't use that.

> If we don't want to worry about the CODEOWNERS problem it might be easiest to use the lgtm plugin configured to respect github approve reviews and have Tide require that label?

The current situation is, in fact, a *non-applicable CODEOWNERS file* scenario. I am looking into the GitHub query (at past, approved PRs) to see what happens in the case of an approval from a non-CODEOWNER. If needed I will take your suggestion and add the `lgtm` plugin (and label), Thanks!

**Review by cjwagner — APPROVED** [review]
Tide probably won't merge anything until the CODEOWNERS thing is addressed, but this should now be safe to merge during oncall hours.

File: prow/config.yaml
Comment (howardjohn) [reply prow/config.yaml:318]: I think you can actually turn off "require codeowners approval". Then anyone can approve even if they aren't a codeowners, which is probably relevant here

Comment (clarketm) [reply prow/config.yaml:318]: > I think you can actually turn off "require codeowners approval". Then anyone can approve even if they aren't a codeowners, which is probably relevant here

Good idea. We should try this first. Let me change that in the `branch-protection` settings and update the PR.

**Comment by clarketm** [comment]
> Tide probably won't merge anything until the CODEOWNERS thing is addressed, but this should now be safe to merge during oncall hours.

I updated the PR to disable CODEOWNERS in branch priotection setting (i.e. `require_code_owner_reviews: false`). I also verified that `review:approved` GitHub query matches the relevant PRs that are approved.

**Review by cjwagner — APPROVED** [review]

**Review by fejta — APPROVED** [review]

**Pull request merged**
